compiler/function-colors.h

```diff
@@ -0,0 +1,306 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2020 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#pragma once
+
+#include <array>
+#include <map>
+#include <vector>
+
+#include <common/termformat/termformat.h>
+#include <compiler/stage.h>
+
+// function_palette namespace contains all the necessary things to
+// implement the concept of colors for functions.
+namespace function_palette {
+
+// color_t enumeration describes all possible colors.
+enum class color_t {
```

Comment:
I don't think that we use `_t` suffix convention.
(Or if we do somewhere, it's probably a legacy code.)
I'm not sure about `Color` vs `color` (upper camel case or just camel case); maybe someone else can suggest something. But aside from that, consider using a enum named `FunctionColor` instead of namespace+Color enum; but @AlexK0 may give more details on that, whether it's OK to make a new namespace here.

Reply:
Now `color_t` is an alias for `uint64_t`, so it's not a class, but a type, so the suffix is probably needed here.